commands/webhook.go

```diff
@@ -0,0 +1,472 @@
+		if channel != nil {
+			outgoingWebhook.ChannelId = channel.Id
+		}
+	}
+
+	createdOutgoing, respWebhookOutgoing := c.CreateOutgoingWebhook(outgoingWebhook)
+	if respWebhookOutgoing.Error != nil {
+		return respWebhookOutgoing.Error
+	}
+
+	printer.Print("Id: " + createdOutgoing.Id)
+	printer.Print("Display Name: " + createdOutgoing.DisplayName)
+
+	return nil
+}
+
+func modifyOutgoingWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	if len(args) < 1 {
+		return errors.New("WebhookID is not specified")
+	}
+
+	webhookArg := args[0]
+	oldHook, respWebhookOutgoing := c.GetOutgoingWebhook(webhookArg)
+	if respWebhookOutgoing.Error != nil {
+		return fmt.Errorf("unable to find webhook '%s'", webhookArg)
+	}
+
+	updatedHook := model.OutgoingWebhookFromJson(strings.NewReader(oldHook.ToJson()))
+
+	channelArg, _ := command.Flags().GetString("channel")
+	if channelArg != "" {
+		channel := getChannelFromChannelArg(c, channelArg)
+		if channel == nil {
+			return fmt.Errorf("unable to find channel '%s'", channelArg)
+		}
+		updatedHook.ChannelId = channel.Id
+	}
+
+	displayName, _ := command.Flags().GetString("display-name")
+	if displayName != "" {
+		updatedHook.DisplayName = displayName
+	}
+
+	description, _ := command.Flags().GetString("description")
+	if description != "" {
+		updatedHook.Description = description
+	}
+
+	triggerWords, err := command.Flags().GetStringArray("trigger-word")
+	if err != nil {
+		return errors.Wrap(err, "invalid trigger-word parameter")
+	}
+	if len(triggerWords) > 0 {
+		updatedHook.TriggerWords = triggerWords
+	}
+
+	triggerWhenString, _ := command.Flags().GetString("trigger-when")
+	if triggerWhenString != "" {
+		var triggerWhen int
+		if triggerWhenString == "exact" {
+			triggerWhen = 0
+		} else if triggerWhenString == "start" {
+			triggerWhen = 1
+		} else {
+			return errors.New("invalid trigger-when parameter")
+		}
+		updatedHook.TriggerWhen = triggerWhen
+	}
+
+	iconURL, _ := command.Flags().GetString("icon")
+	if iconURL != "" {
+		updatedHook.IconURL = iconURL
+	}
+
+	contentType, _ := command.Flags().GetString("content-type")
+	if contentType != "" {
+		updatedHook.ContentType = contentType
+	}
+
+	callbackURLs, err := command.Flags().GetStringArray("url")
+	if err != nil {
+		return errors.Wrap(err, "invalid URL parameter")
+	}
+	if len(callbackURLs) > 0 {
+		updatedHook.CallbackURLs = callbackURLs
+	}
+
+	if _, response := c.UpdateOutgoingWebhook(updatedHook); response.Error != nil {
+		return response.Error
+	}
+
+	return nil
+}
+
+func deleteWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	if len(args) < 1 {
+		return errors.New("WebhookID is not specified")
+	}
+
+	webhookId := args[0]
+	_, respIncomingWebhook := c.DeleteIncomingWebhook(webhookId)
+	_, respOutgoingWebhook := c.DeleteOutgoingWebhook(webhookId)
+
+	if respIncomingWebhook.Error != nil && respOutgoingWebhook.Error != nil {
```

Comment:
I'm not 100% convinced on the approach of trying to delete both and erroring if one fails. Would you be up to modify this to trying to get both incoming and outgoing webhooks for the given ID, erroring if there is none and deleting the one that matches?